clang/test/Analysis/array-init-loop.cpp

```diff
@@ -330,3 +330,41 @@ void no_crash() {
 }
 
 } // namespace crash
+
+namespace array_subscript_initializer {
+struct S {
+  int x;
+};
+
+void no_crash() {
+  S arr[][2] = {{1, 2}};
+
+  const auto [a, b] = arr[0]; // no-crash
```

Comment:
@steakhal `// no-crash` at the end of the line. Similarly in the other test case.